src/tzdb.rs

```diff
@@ -90,11 +116,11 @@ impl Tzif {
     }
 
     // There are various other ways to search rather than binary_search. See glibc
-    pub fn binary_search(&self, epoch_seconds: &Seconds) -> Option<Result<usize, usize>> {
-        self.0
-            .data_block2
-            .as_ref()
-            .map(|b| b.transition_times.binary_search(epoch_seconds))
+    pub fn binary_search(&self, epoch_seconds: &Seconds) -> TemporalResult<Result<usize, usize>> {
+        Ok(self
+            .get_data_block2()?
+            .transition_times
+            .binary_search(epoch_seconds))
```

Comment:
My suggestion was more on the lines of removing `binary_search` and just doing this when doing the binary search:
```
let db = self.get_data_block2()?;
let result = db.transition_times.binary_search(epoch_seconds);

Reply:
Ooooooh, we definitely can. [Glibc](https://codebrowser.dev/glibc/glibc/time/tzfile.c.html#__tzfile_compute) does a somewhat gnarly combination of searching that I was intrigued in seeing if it was actually more optimized. Think I was being a bit overly optimistic in separating that out early :sweat_smile: